src/server/resolvers.ts

```diff
@@ -27,6 +27,7 @@ async function queryTerms(object: any, args: any, context: any): Promise<any> {
       (distributionIri: string) => new IRI(distributionIri)
     ),
     query: args.query,
+    timeoutMs: parseInt(process.env.QUERY_TIMEOUT as string) || 10000,
```

Comment:
This means the timeout is configured on the server: clients have no way to provide their own timeout. I decided for this over the alternative where clients can provide their own desired timeout in queries for two reasons: 

1. I don’t want to burden clients with choosing some timeout.
2. We would still need some server-side upper bound so clients cannot have extremely long timeouts (e.g. 1000 sec).

However, this comes at the cost of some flexibility for clients. Should we assume they can all live with our default (10 sec) or do individual clients really need to set their own timeouts?

Reply:
Yes, good point!

I'm not sure if all clients can live with our default. Perhaps some clients are willing to wait longer in case of slow(er) sources. That argues for a configuration option on the client. If that makes sense, we could even lower our default to, say, 3 to 5 seconds.

Reply:
Added a `timeoutMs` GraphQL query param.